pkg/policy/groups/actions.go

```diff
@@ -85,10 +105,42 @@ func UpdateDerivativeCNPIfNeeded(newCNP *cilium_v2.CiliumNetworkPolicy, oldCNP *
 		return false
 	}
 
-	controllerManager.UpdateController(fmt.Sprintf("CNP-Derivative-update-%s", newCNP.ObjectMeta.Name),
+	controllerManager.UpdateController(fmt.Sprintf("update-derivative-cnp-%s", newCNP.ObjectMeta.Name),
 		controller.ControllerParams{
 			DoFunc: func(ctx context.Context) error {
-				return addDerivativeCNP(ctx, newCNP)
+				return addDerivativePolicy(ctx, newCNP, false)
+			},
+		})
+	return true
+}
+
+// UpdateDerivativeCCNPIfNeeded updates or creates a CCNP if the given CCNP has
+// any rule that needs to create a new derivative policy(eg: ToGroups). In case
+// that the new CCNP does not have any derivative policy and the old one had
+// one, it will delete the old policy.
```

Comment:
I think we can document when this returns `true` or `false`. Can you update the other similar functions as well?